postgrid/__init__.py

```diff
@@ -470,6 +479,118 @@ def construct_event(cls, payload, secret):
             raise ValueError()
 
 
+def _av_post_request(endpoint, json=None, data=None, params=None):
+    data = _map_keys_recursive(data, _snake_to_camel)
+    params = _map_keys_recursive(params, _snake_to_camel)
+
+    headers = {'x-api-key': av_key}
+    data_to_pass = []
+    params_to_pass = []
+
+    def flatten(arr, key, value):
+        if isinstance(value, dict):
+            for inner_key, inner_value in value.items():
+                flatten(arr, f'{key}[{inner_key}]', inner_value)
+        elif isinstance(value, list):
+            for inner_value in value:
+                flatten(arr, f'{key}[]', inner_value)
+        elif isinstance(value, bool):
+            arr.append((key, 'true' if value else 'false'))
+        elif value is not None:
+            arr.append((key, value))
+
+    if data:
+        for key, value in data.items():
+            flatten(data_to_pass, key, value)
+
+    if params:
+        for key, value in params.items():
+            flatten(params_to_pass, key, value)
+
+    res = requests.post(
+        'https://api.postgrid.com/v1/addver' + endpoint,
+        json=json,
+        data=data_to_pass,
+        params=params_to_pass,
+        headers=headers,
+    )
+    value = res.json()
+
+    try:
+        res.raise_for_status()
+    except requests.HTTPError as e:
+        raise AVError(status_code=res.status_code, message=value['message'])
+    return _av_convert_json_value(value)
+
+class Address(BaseResource):
+    endpoint = 'addver'
+
+    @classmethod
+    def lookup_info(cls):
+        return _request(av_base, 'addver', method='GET')
+
+    @classmethod
+    def verify(cls, address=None):
+        return _request(av_base, 'addver/verifications', method='POST', **locals())
+
+    @classmethod
+    def autocomplete_previews(
+        cls, partial_street=None, country_filter=None, prov_instead_of_pc=None
+    ):
+        return _request(av_base, 'addver/completions', method='GET', **locals())
+
+    @classmethod
+    def autocomplete_address(
```

Comment:
The index is an optional parameter in case the user wants to get the full list of complete addresses

Reply:
Fixed.